### ISSUE TITLE
sendBatch not sending multiple batches for the same topic

### ISSUE THREAD

**Issue by benzaita** [tulios/kafkajs#622]
I’ve hit something which feels like a bug. Using `sendBatch` to send this works perfectly:

```
const topicMessages =
    [
        {
            "topic": "topic",
            "messages": [
                {
                    "value": "message1"
                },
                {
                    "value": "message2"
                }
            ]
        }
    ]
```

However, sending this ends up producing only the second message:
```
const topicMessages =
    [
        {
            "topic": "topic",
            "messages": [
                {
                    "value": "message1"
                }
            ]
        },
        {
            "topic": "topic",
            "messages": [
                {
                    "value": "message2"
                }
            ]
        }
    ]
```

Is this expected?

**Comment by Nevon** [comment]
Although we never intended to let you specify multiple `TopicMessages` objects for the same topic, the current behavior is not exactly expected.

As I see it, we either:

a) Throw some kind of validation error

b) Do a pass over the input and consolidate all the messages for the same topic.

I think I prefer b, because it's the least surprising thing and it will "Just Work" without the developer having to do anything. One could argue that it's most likely a mistake by the developer if they are including the same topic several times, but I could see how if you're doing some kind of windowing of event transformations over time, a naive implementation might just transform each incoming event into its corresponding outgoing event and end up with multiple `TopicMessages` for the same topic.

**Comment by Nevon** [comment]
Ended up going with option b. This is now available in `1.12.0-beta.20`

**Comment by AlexRex** [comment]
I was getting crazy regarding this one. Thanks for pointing it out! 😄
